### DISCUSSION TITLE
Eclipse GlassFish 8 will support just Java21+

### DISCUSSION THREAD

**Discussion by dmatej** [eclipse-ee4j/glassfish#24790] [Polls]
As there is still ongoing discussion about supported Java versions of Jakarta EE 11, we are asking our community about the opinion, especially active committers.

GlassFish 7 supports Java 11+, preferred is Java 17. Implements Jakarta EE 10.

What about GlassFish 8? Can you add your thoughts?

Results are not obligatory, but we want to know your opinion.

**Comment by dmatej** [comment]
My opinion:

For 17: To have the widest support as possible.

Against 17: 
* To have the widest support as possible means shooting on too many targets which means resources which could be used to improve the overall quality, lower efficiency, more conditions, compromises to support obsoleted Java versions, etc. 
* It means also confusing our community - I don't remember how many times we had to explain which Java version users and contributors should use, when and why.
* GF8 is a next generation, while predecessor GF7 already preferred Java17 (while it still supported Java11).

**Comment by hs536** [comment]
My opinion is as posted on the platform's issue:
https://github.com/jakartaee/platform/issues/820#issuecomment-1888838824

and we can confirm the voting status of the spec committee in the following archive:
https://www.eclipse.org/lists/jakarta.ee-spec/msg03173.html

**Comment by avpinchuk** [comment]
Supporting of two LTS versions at the same time will lead to increased efforts to maintain the code base. This is a step back I think.

**Comment by Emily-Jiang** [comment]
@dmatej the way you designed the vote only allows for one version to be chosen while the question is about plural JDK versions. Will you be able redo the question to use the boxes so that the choice of both JDKs are available?

**Reply by dmatej** [reply]
Actually that is not true, it is why there is the `+` sign. It means 17 and later, 21 and later releases. 

As we wrote in comments, the problem is the amount of additional work for GF team, while nobody else would contribute anything and there is no advantage for GlassFish users (which are usually those behind GF contributors).

**Comment by dmatej** [comment]
Final solution - GlassFish 8 will be based on JDK21, and we will release special version based on JDK17 which will pass all TCK tests, but will not be recommended for production use.
